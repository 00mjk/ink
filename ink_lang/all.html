<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="List of all items in this crate"><meta name="keywords" content="rust, rustlang, rust-lang"><title>List of all items in this crate</title><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../dark.css"><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script src="../storage.js"></script><noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="shortcut icon" href="../favicon.ico"><style type="text/css">#crate-search{background-image:url("../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../ink_lang/index.html'><div class='logo-container'><img src='../rust-logo.png' alt='logo'></div></a></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><a id="settings-menu" href="../settings.html"><img src="../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class='fqn'><span class='out-of-band'><span id='render-detail'><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class='inner'>&#x2212;</span>]</a></span>
        </span>
        <span class='in-band'>List of all items</span></h1><h3 id='Structs'>Structs</h3><ul class='structs docblock'><li><a href='struct.DispatchRetCode.html'>DispatchRetCode</a></li><li><a href='struct.EnvAccess.html'>EnvAccess</a></li></ul><h3 id='Enums'>Enums</h3><ul class='enums docblock'><li><a href='enum.DispatchError.html'>DispatchError</a></li><li><a href='enum.DispatchMode.html'>DispatchMode</a></li></ul><h3 id='Traits'>Traits</h3><ul class='traits docblock'><li><a href='trait.BaseEvent.html'>BaseEvent</a></li><li><a href='trait.Constructor.html'>Constructor</a></li><li><a href='trait.ConstructorDispatcher.html'>ConstructorDispatcher</a></li><li><a href='trait.DispatchUsingMode.html'>DispatchUsingMode</a></li><li><a href='trait.EmitEvent.html'>EmitEvent</a></li><li><a href='trait.Env.html'>Env</a></li><li><a href='trait.Execute.html'>Execute</a></li><li><a href='trait.FnInput.html'>FnInput</a></li><li><a href='trait.FnOutput.html'>FnOutput</a></li><li><a href='trait.FnSelector.html'>FnSelector</a></li><li><a href='trait.FnState.html'>FnState</a></li><li><a href='trait.ForwardCall.html'>ForwardCall</a></li><li><a href='trait.ForwardCallMut.html'>ForwardCallMut</a></li><li><a href='trait.MessageDispatcher.html'>MessageDispatcher</a></li><li><a href='trait.MessageMut.html'>MessageMut</a></li><li><a href='trait.MessageRef.html'>MessageRef</a></li><li><a href='trait.StaticEnv.html'>StaticEnv</a></li><li><a href='trait.ToAccountId.html'>ToAccountId</a></li></ul><h3 id='Attribute Macros'>Attribute Macros</h3><ul class='attributes docblock'><li><a href='attr.contract.html'>contract</a></li><li><a href='attr.test.html'>test</a></li></ul><h3 id='Functions'>Functions</h3><ul class='functions docblock'><li><a href='fn.execute_constructor.html'>execute_constructor</a></li><li><a href='fn.execute_message.html'>execute_message</a></li><li><a href='fn.execute_message_mut.html'>execute_message_mut</a></li></ul><h3 id='Typedefs'>Typedefs</h3><ul class='typedefs docblock'><li><a href='type.DispatchResult.html'>DispatchResult</a></li></ul></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../";window.currentCrate = "ink_lang";</script><script src="../main.js"></script><script defer src="../search-index.js"></script></body></html>