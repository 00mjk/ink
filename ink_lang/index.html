<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `ink_lang` crate."><meta name="keywords" content="rust, rustlang, rust-lang, ink_lang"><title>ink_lang - Rust</title><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../ayu.css" disabled ><script src="../storage.js"></script><noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../favicon.svg">
<link rel="alternate icon" type="image/png" href="../favicon-16x16.png">
<link rel="alternate icon" type="image/png" href="../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../ink_lang/index.html'><div class='logo-container rust-logo'><img src='../rust-logo.png' alt='logo'></div></a><p class='location'>Crate ink_lang</p><div class='block version'><p>Version 3.0.0</p></div><div class="sidebar-elems"><a id='all-types' href='all.html'><p>See all ink_lang's items</p></a><div class="block items"><ul><li><a href="#reexports">Re-exports</a></li><li><a href="#structs">Structs</a></li><li><a href="#traits">Traits</a></li></ul></div><p class='location'></p><script>window.sidebarCurrent = {name: 'ink_lang', ty: 'mod', relpath: '../'};</script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><span class="help-button">?</span>
                <a id="settings-menu" href="../settings.html"><img src="../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class='fqn'><span class='out-of-band'><span id='render-detail'><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class='inner'>&#x2212;</span>]</a></span><a class='srclink' href='../src/ink_lang/lib.rs.html#15-80' title='goto source code'>[src]</a></span><span class='in-band'>Crate <a class="mod" href=''>ink_lang</a></span></h1><h2 id='reexports' class='section-header'><a href="#reexports">Re-exports</a></h2>
<table><tr><td><code>pub use <a class="mod" href="https://docs.rs/static_assertions/1.1.0/static_assertions/index.html" title="mod static_assertions">static_assertions</a>;</code></td></tr></table><h2 id='structs' class='section-header'><a href="#structs">Structs</a></h2>
<table><tr class='module-item'><td><a class="struct" href="struct.EnvAccess.html" title='ink_lang::EnvAccess struct'>EnvAccess</a></td><td class='docblock-short'><p>A typed accessor to the environment.</p>
</td></tr></table><h2 id='traits' class='section-header'><a href="#traits">Traits</a></h2>
<table><tr class='module-item'><td><a class="trait" href="trait.BaseEvent.html" title='ink_lang::BaseEvent trait'>BaseEvent</a></td><td class='docblock-short'><p>Defines a base event type for the contract.</p>
</td></tr><tr class='module-item'><td><a class="trait" href="trait.ContractEnv.html" title='ink_lang::ContractEnv trait'>ContractEnv</a></td><td class='docblock-short'><p>The environment of the compiled ink! smart contract.</p>
</td></tr><tr class='module-item'><td><a class="trait" href="trait.EmitEvent.html" title='ink_lang::EmitEvent trait'>EmitEvent</a></td><td class='docblock-short'><p>Implemented by contracts in order to override <code>env().emit_event(..)</code>
syntax for emitting of ink! contract events.</p>
</td></tr><tr class='module-item'><td><a class="trait" href="trait.Env.html" title='ink_lang::Env trait'>Env</a></td><td class='docblock-short'><p>Simplifies interaction with the host environment via <code>self</code>.</p>
</td></tr><tr class='module-item'><td><a class="trait" href="trait.StaticEnv.html" title='ink_lang::StaticEnv trait'>StaticEnv</a></td><td class='docblock-short'><p>Simplifies interaction with the host environment via <code>Self</code>.</p>
</td></tr><tr class='module-item'><td><a class="trait" href="trait.ToAccountId.html" title='ink_lang::ToAccountId trait'>ToAccountId</a></td><td class='docblock-short'><p>Implemented by contracts that are compiled as dependencies.</p>
</td></tr></table><h2 id='attributes' class='section-header'><a href="#attributes">Attribute Macros</a></h2>
<table><tr class='module-item'><td><a class="attr" href="attr.contract.html" title='ink_lang::contract attr'>contract</a></td><td class='docblock-short'></td></tr><tr class='module-item'><td><a class="attr" href="attr.test.html" title='ink_lang::test attr'>test</a></td><td class='docblock-short'></td></tr><tr class='module-item'><td><a class="attr" href="attr.trait_definition.html" title='ink_lang::trait_definition attr'>trait_definition</a></td><td class='docblock-short'></td></tr></table></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../";window.currentCrate = "ink_lang";</script><script src="../main.js"></script><script defer src="../search-index.js"></script></body></html>