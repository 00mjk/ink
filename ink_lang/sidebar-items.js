initSidebarItems({"attr":[["contract",""]],"enum":[["BuildPhase","Building phase of a contract construction."],["DispatchError","A dispatch error."],["DispatchMode","The contract dispatch mode."],["EmptyPhase","Phase to start the building process of a contract."],["FinalPhase","Final phase of a fully constructed contract."]],"struct":[["Contract","The contract definition."],["DispatchList","A list of dispatchers."],["DispatchRetCode","A return code indicating success or error in a compact form."],["EnvAccess","A typed accessor to the environment."],["MsgCon","A constructor contract message wrapper."],["MsgMut","A `&mut self` contract message wrapper."],["MsgRef","A `&self` contract message wrapper."],["Placeholder","Placeholder for the given type."],["UnreachableDispatcher","A dispatcher that shall never dispatch."]],"trait":[["BaseEvent","Defines a base event type for the contract."],["Constructor","A dispatchable contract constructor message."],["Dispatch","Types implementing this trait can handle contract calls."],["DispatchUsingMode","Trait implemented by contracts themselves in order to provide a clean interface for the C-ABI specified `call` and `create` functions to forward calls to."],["EmitEvent","Implemented by contracts in order to override `env().emit_event(..)` syntax for emitting of ink! contract events."],["Env","Simplifies interaction with the host environment via `self`."],["FnInput","Dispatchable functions that have inputs."],["FnOutput","Dispatchable functions that have an output."],["FnSelector","The selector of dispatchable functions."],["FnState","The storage state that the dispatchable function acts on."],["ForwardCall","Implemented by contracts that are compiled as dependencies."],["ForwardCallMut","Implemented by contracts that are compiled as dependencies."],["GenerateAbi","Types implementing this trait can generate their metadata."],["InstantiateTestable","Trait implemented by contracts to make them testable."],["MessageMut","A `&mut self` dispatchable contract message."],["MessageRef","A `&self` dispatchable contract message."],["PushDispatcher","Types able to push another dispatcher to themselves."],["StaticEnv","Simplifies interaction with the host environment via `Self`."],["ToAccountId","Implemented by contracts that are compiled as dependencies."]],"type":[["DispatchResult","A dispatch result."],["EmptyDispatchList","A simple type definition to view the single unreachable dispatcher as list."]]});