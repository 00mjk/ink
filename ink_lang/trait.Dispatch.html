<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `Dispatch` trait in crate `ink_lang`."><meta name="keywords" content="rust, rustlang, rust-lang, Dispatch"><title>ink_lang::Dispatch - Rust</title><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../dark.css"><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script src="../storage.js"></script><noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="shortcut icon" href="../favicon.ico"><style type="text/css">#crate-search{background-image:url("../down-arrow.svg");}</style></head><body class="rustdoc trait"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../ink_lang/index.html'><div class='logo-container'><img src='../rust-logo.png' alt='logo'></div></a><p class='location'>Trait Dispatch</p><div class="sidebar-elems"><div class="block items"><a class="sidebar-title" href="#required-methods">Required Methods</a><div class="sidebar-links"><a href="#tymethod.dispatch">dispatch</a></div><a class="sidebar-title" href="#implementors">Implementors</a></div><p class='location'><a href='index.html'>ink_lang</a></p><script>window.sidebarCurrent = {name: 'Dispatch', ty: 'trait', relpath: ''};</script><script defer src="sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><a id="settings-menu" href="../settings.html"><img src="../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class='fqn'><span class='out-of-band'><span id='render-detail'><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class='inner'>&#x2212;</span>]</a></span><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#50-53' title='goto source code'>[src]</a></span><span class='in-band'>Trait <a href='index.html'>ink_lang</a>::<wbr><a class="trait" href=''>Dispatch</a></span></h1><div class="docblock type-decl hidden-by-usual-hider"><pre class='rust trait'>pub trait Dispatch {
    fn <a href='#tymethod.dispatch' class='fnname'>dispatch</a>(input: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;;
}</pre></div><div class='docblock'><p>Types implementing this trait can handle contract calls.</p>
</div>
            <h2 id='required-methods' class='small-section-header'>Required methods<a href='#required-methods' class='anchor'></a></h2><div class='methods'><h3 id='tymethod.dispatch' class='method'><code id='dispatch.v'>fn <a href='#tymethod.dispatch' class='fnname'>dispatch</a>(input: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;</code></h3><div class='docblock'><p>Dispatches the call and returns the call result.</p>
</div></div><span class='loading-content'>Loading content...</span>
            <h2 id='implementors' class='small-section-header'>Implementors<a href='#implementors' class='anchor'></a></h2><div class='item-list' id='implementors-list'><h3 id='impl-Dispatch' class='impl'><code class='in-band'>impl Dispatch for <a class="struct" href="../ink_lang/struct.UnreachableDispatcher.html" title="struct ink_lang::UnreachableDispatcher">UnreachableDispatcher</a></code><a href='#impl-Dispatch' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#64-69' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='method.dispatch' class="method hidden"><code id='dispatch.v-1'>fn <a href='#method.dispatch' class='fnname'>dispatch</a>(_data: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;</code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#66-68' title='goto source code'>[src]</a></h4></div><h3 id='impl-Dispatch-1' class='impl'><code class='in-band'>impl&lt;Head, Rest&gt; Dispatch for <a class="struct" href="../ink_lang/struct.DispatchList.html" title="struct ink_lang::DispatchList">DispatchList</a>&lt;Head, Rest&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;Head: <a class="trait" href="../ink_lang/trait.Dispatch.html" title="trait ink_lang::Dispatch">Dispatch</a> + <a class="trait" href="../ink_lang/trait.FnSelector.html" title="trait ink_lang::FnSelector">FnSelector</a>,<br>&nbsp;&nbsp;&nbsp;&nbsp;Rest: <a class="trait" href="../ink_lang/trait.Dispatch.html" title="trait ink_lang::Dispatch">Dispatch</a>,&nbsp;</span></code><a href='#impl-Dispatch-1' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#119-132' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='method.dispatch-1' class="method hidden"><code id='dispatch.v-2'>fn <a href='#method.dispatch-1' class='fnname'>dispatch</a>(data: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;</code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#125-131' title='goto source code'>[src]</a></h4></div><h3 id='impl-Dispatch-2' class='impl'><code class='in-band'>impl&lt;M&gt; Dispatch for <a class="struct" href="../ink_lang/struct.MsgCon.html" title="struct ink_lang::MsgCon">MsgCon</a>&lt;M&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;M: <a class="trait" href="../ink_lang/trait.Constructor.html" title="trait ink_lang::Constructor">Constructor</a>,&nbsp;</span></code><a href='#impl-Dispatch-2' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#227-243' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='method.dispatch-2' class="method hidden"><code id='dispatch.v-3'>fn <a href='#method.dispatch-2' class='fnname'>dispatch</a>(data: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;</code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#232-242' title='goto source code'>[src]</a></h4></div><h3 id='impl-Dispatch-3' class='impl'><code class='in-band'>impl&lt;M&gt; Dispatch for <a class="struct" href="../ink_lang/struct.MsgMut.html" title="struct ink_lang::MsgMut">MsgMut</a>&lt;M&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;M: <a class="trait" href="../ink_lang/trait.MessageMut.html" title="trait ink_lang::MessageMut">MessageMut</a>,&nbsp;</span></code><a href='#impl-Dispatch-3' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#188-209' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='method.dispatch-3' class="method hidden"><code id='dispatch.v-4'>fn <a href='#method.dispatch-3' class='fnname'>dispatch</a>(data: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;</code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#193-208' title='goto source code'>[src]</a></h4></div><h3 id='impl-Dispatch-4' class='impl'><code class='in-band'>impl&lt;M&gt; Dispatch for <a class="struct" href="../ink_lang/struct.MsgRef.html" title="struct ink_lang::MsgRef">MsgRef</a>&lt;M&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;M: <a class="trait" href="../ink_lang/trait.MessageRef.html" title="trait ink_lang::MessageRef">MessageRef</a>,&nbsp;</span></code><a href='#impl-Dispatch-4' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#150-170' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='method.dispatch-4' class="method hidden"><code id='dispatch.v-5'>fn <a href='#method.dispatch-4' class='fnname'>dispatch</a>(data: &amp;<a class="struct" href="../ink_core/env/call/utils/struct.CallData.html" title="struct ink_core::env::call::utils::CallData">CallData</a>) -&gt; <a class="enum" href="https://doc.rust-lang.org/nightly/core/result/enum.Result.html" title="enum core::result::Result">Result</a>&lt;<a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.unit.html">()</a>, <a class="enum" href="../ink_lang/enum.DispatchError.html" title="enum ink_lang::DispatchError">DispatchError</a>&gt;</code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#155-169' title='goto source code'>[src]</a></h4></div></div><span class='loading-content'>Loading content...</span><script type="text/javascript" src="../implementors/ink_lang/trait.Dispatch.js" async></script></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../";window.currentCrate = "ink_lang";</script><script src="../main.js"></script><script defer src="../search-index.js"></script></body></html>