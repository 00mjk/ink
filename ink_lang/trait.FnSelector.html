<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="API documentation for the Rust `FnSelector` trait in crate `ink_lang`."><meta name="keywords" content="rust, rustlang, rust-lang, FnSelector"><title>ink_lang::FnSelector - Rust</title><link rel="stylesheet" type="text/css" href="../normalize.css"><link rel="stylesheet" type="text/css" href="../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../dark.css"><link rel="stylesheet" type="text/css" href="../light.css" id="themeStyle"><script src="../storage.js"></script><noscript><link rel="stylesheet" href="../noscript.css"></noscript><link rel="shortcut icon" href="../favicon.ico"><style type="text/css">#crate-search{background-image:url("../down-arrow.svg");}</style></head><body class="rustdoc trait"><!--[if lte IE 8]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu">&#9776;</div><a href='../ink_lang/index.html'><div class='logo-container'><img src='../rust-logo.png' alt='logo'></div></a><p class='location'>Trait FnSelector</p><div class="sidebar-elems"><div class="block items"><a class="sidebar-title" href="#associated-const">Associated Constants</a><div class="sidebar-links"><a href="#associatedconstant.SELECTOR">SELECTOR</a></div><a class="sidebar-title" href="#implementors">Implementors</a></div><p class='location'><a href='index.html'>ink_lang</a></p><script>window.sidebarCurrent = {name: 'FnSelector', ty: 'trait', relpath: ''};</script><script defer src="sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!"><img src="../brush.svg" width="18" alt="Pick another theme!"></button><div id="theme-choices"></div></div><script src="../theme.js"></script><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input" name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><a id="settings-menu" href="../settings.html"><img src="../wheel.svg" width="18" alt="Change settings"></a></div></form></nav><section id="main" class="content"><h1 class='fqn'><span class='out-of-band'><span id='render-detail'><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class='inner'>&#x2212;</span>]</a></span><a class='srclink' href='../src/ink_lang/traits.rs.html#33-36' title='goto source code'>[src]</a></span><span class='in-band'>Trait <a href='index.html'>ink_lang</a>::<wbr><a class="trait" href=''>FnSelector</a></span></h1><div class="docblock type-decl hidden-by-usual-hider"><pre class='rust trait'>pub trait FnSelector {
    const <a href='#associatedconstant.SELECTOR' class="constant"><b>SELECTOR</b></a>: <a class="struct" href="../ink_core/env/call/utils/struct.Selector.html" title="struct ink_core::env::call::utils::Selector">Selector</a>;
}</pre></div><div class='docblock'><p>The selector of dispatchable functions.</p>
</div>
            <h2 id='associated-const' class='small-section-header'>Associated Constants<a href='#associated-const' class='anchor'></a></h2><div class='methods'><h3 id='associatedconstant.SELECTOR' class='method'><code id='SELECTOR.v'>const <a href='#associatedconstant.SELECTOR' class="constant"><b>SELECTOR</b></a>: <a class="struct" href="../ink_core/env/call/utils/struct.Selector.html" title="struct ink_core::env::call::utils::Selector">Selector</a></code></h3><div class='docblock'><p>The selector.</p>
</div></div><span class='loading-content'>Loading content...</span>
            <h2 id='implementors' class='small-section-header'>Implementors<a href='#implementors' class='anchor'></a></h2><div class='item-list' id='implementors-list'><h3 id='impl-FnSelector' class='impl'><code class='in-band'>impl&lt;M&gt; FnSelector for <a class="struct" href="../ink_lang/struct.MsgCon.html" title="struct ink_lang::MsgCon">MsgCon</a>&lt;M&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;M: <a class="trait" href="../ink_lang/trait.Constructor.html" title="trait ink_lang::Constructor">Constructor</a>,&nbsp;</span></code><a href='#impl-FnSelector' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#220-225' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='associatedconstant.SELECTOR-1' class="associatedconstant hidden"><code id='SELECTOR.v-1'>const <a href='#associatedconstant.SELECTOR-1' class="constant"><b>SELECTOR</b></a>: <a class="struct" href="../ink_core/env/call/utils/struct.Selector.html" title="struct ink_core::env::call::utils::Selector">Selector</a></code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#224' title='goto source code'>[src]</a></h4></div><h3 id='impl-FnSelector-1' class='impl'><code class='in-band'>impl&lt;M&gt; FnSelector for <a class="struct" href="../ink_lang/struct.MsgMut.html" title="struct ink_lang::MsgMut">MsgMut</a>&lt;M&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;M: <a class="trait" href="../ink_lang/trait.MessageMut.html" title="trait ink_lang::MessageMut">MessageMut</a>,&nbsp;</span></code><a href='#impl-FnSelector-1' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#181-186' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='associatedconstant.SELECTOR-2' class="associatedconstant hidden"><code id='SELECTOR.v-2'>const <a href='#associatedconstant.SELECTOR-2' class="constant"><b>SELECTOR</b></a>: <a class="struct" href="../ink_core/env/call/utils/struct.Selector.html" title="struct ink_core::env::call::utils::Selector">Selector</a></code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#185' title='goto source code'>[src]</a></h4></div><h3 id='impl-FnSelector-2' class='impl'><code class='in-band'>impl&lt;M&gt; FnSelector for <a class="struct" href="../ink_lang/struct.MsgRef.html" title="struct ink_lang::MsgRef">MsgRef</a>&lt;M&gt; <span class="where fmt-newline">where<br>&nbsp;&nbsp;&nbsp;&nbsp;M: <a class="trait" href="../ink_lang/trait.MessageRef.html" title="trait ink_lang::MessageRef">MessageRef</a>,&nbsp;</span></code><a href='#impl-FnSelector-2' class='anchor'></a><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#143-148' title='goto source code'>[src]</a></h3><div class='impl-items'><h4 id='associatedconstant.SELECTOR-3' class="associatedconstant hidden"><code id='SELECTOR.v-3'>const <a href='#associatedconstant.SELECTOR-3' class="constant"><b>SELECTOR</b></a>: <a class="struct" href="../ink_core/env/call/utils/struct.Selector.html" title="struct ink_core::env::call::utils::Selector">Selector</a></code><a class='srclink' href='../src/ink_lang/dispatcher.rs.html#147' title='goto source code'>[src]</a></h4></div></div><span class='loading-content'>Loading content...</span><script type="text/javascript" src="../implementors/ink_lang/trait.FnSelector.js" async></script></section><section id="search" class="content hidden"></section><section class="footer"></section><script>window.rootPath = "../";window.currentCrate = "ink_lang";</script><script src="../main.js"></script><script defer src="../search-index.js"></script></body></html>