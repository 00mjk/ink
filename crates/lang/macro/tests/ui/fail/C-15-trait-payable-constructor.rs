use ink_lang as ink;

#[ink::trait_definition]
pub trait Foo {
    #[ink(constructor)]
    fn new() -> Self;
}

#[ink::contract]
mod noop {
    #[ink(storage)]
    pub struct Noop {}

    impl Foo for Noop {
        #[ink(constructor, payable)]
        fn new() -> Self {
            Self {}
        }
    }
}

fn main() {}
