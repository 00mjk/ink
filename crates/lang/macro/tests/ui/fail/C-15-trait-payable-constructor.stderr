error: encountered conflicting ink! attribute argument: constructor is implicitly payable
  --> $DIR/C-15-trait-payable-constructor.rs:15:28
   |
15 |         #[ink(constructor, payable)]
   |                            ^^^^^^^
